File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;

//ResponseBody and Controller的结合体
@RestController
@RequestMapping("/brand")
public class BrandController {
	
	@Reference
	private BrandService brandService;
	
	/**
	 * 查询所有
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		return brandService.findAll();
	} 
	
	/**
	 * 普通查询方法,不带检索
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult findPage(int page,int rows){
		return brandService.findPage(page,rows);
	} 
	/**
	 * 替代了上面的findPage方法,这是带检索的查询方法
	 * @param page
	 * @param rows
	 * @param brand
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult findPage(int page,int rows,@RequestBody TbBrand brand){
		return brandService.findPage(page,rows,brand);
	} 
	
	/**
	 * 添加商品
	 * @param brand
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand brand){
		try {
			brandService.add(brand);
			return new Result(true,"Successfully add!");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"Errors had occurred,add failed!");
		}
	} 
	
	/**
	 * 更新商品
	 * @param brand
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand brand){
		try {
			brandService.update(brand);
			return new Result(true,"Successfully updated!");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"Errors had occurred,updated failed!");
		}
	} 
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbBrand findOne(Long id){
		return brandService.findOne(id);
	} 
	
	@RequestMapping("/delete")
	public Result delete(Long[] ids){
		try {
			brandService.delete(ids);
			return new Result(true,"Successfully delete!");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"Errors had occurred,delete failed!");
		}
	} 
	
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList(){
		return brandService.selectOptionList();
	} 


}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
	//品牌控制层
	app.controller('brandController',function($scope,$controller,brandService){
		//伪继承,{$scope:$scope}的 写法  可以实现类似继承的效果
		$controller('baseController',{$scope:$scope});//继承
		//查询所有商品数据
		$scope.findAll=function(){
			brandService.findAll().success(
				function(response){
					$scope.list=response;
				}			
			);
		}
		
		//分页查询方法,和reloadList关联,通过其调用findPage获得参数
		$scope.findPage = function(page,rows){
			brandService.findPage(page,rows).success(
				function(response){
					$scope.list = response.rows;//显示当前数据
					$scope.paginationConf.totalItems = response.total;//更新总记录数
				}
			);
		}
		
		//定义一个搜索体的绑定变量
		$scope.searchEntity={};
		//search方法,代替findPage方法
		$scope.search = function(page,rows){
			brandService.search(page,rows,$scope.searchEntity).success(
				function(response){
					$scope.list = response.rows;
					$scope.paginationConf.totalItems = response.total;
				}
			);
		}
		
		//商品保存以及更新方法
		$scope.save = function(){
			if($scope.entity.id!=null){
				//更新方法
				brandService.update($scope.entity).success(function(response){
					if(response.success){
						//重新查询
						$scope.reloadList();
					} else {
						alert(response.message);
					}
				});
			}else{
				//增加方法
				brandService.add($scope.entity).success(function(response){
					if(response.success){
						//重新查询
						$scope.reloadList();
					} else {
						alert(response.message);
					}
				});
			}
		}
		
		//根据id获取实体
		$scope.findOne = function(id){
			brandService.findOne(id).success(function(response){
				$scope.entity = response;
			});
		}
		
		//批量删除
		$scope.dele = function(){
			if($scope.selectIds.length == 0){
				alert('请至少选择一个进行删除!');
				return;
			}
			//根据选中的id进行删除操作
			brandService.dele($scope.selectIds).success(function(response){
				if(response.success){
					$scope.reloadList();
					//删除成功,清空删除id数组
					$scope.selectIds = [];
				} else{
					alert(response.message);
				}
			});
		}
		
	});